src/Spago/Config.hs

```diff
@@ -373,5 +376,5 @@ transformMExpr rules =
 addDependencies :: Spago m => Config -> [PackageName] -> m ()
 addDependencies config newPackages = do
   configHasChanged <- withConfigAST $ addRawDeps config newPackages
-  when (not configHasChanged) $ do
-    echo "WARNING: configuration file was not updated. You should have a record with the `dependencies` key for this to work."
+  unless configHasChanged $
```

Comment:
The key change: `configHasChanged == False` can happen for multiple reasons:
1. Misconfiguration of spago.dhall (e.g. missing `dependencies` field, or `dependencies` field not having type `List String`)
2. spago.dhall already contains the dep which user tried to install
3. adding empty package list

Fix - make this warning more generic and give more granular warnings in `addRawDeps`

Reply:
Yeah this is the best we cann do for now. I'd like to figure something out so that we can give specific error messages instead of this generic one, but this change is indeed much better than giving the wrong message out 🙂